csi.proto

```diff
@@ -172,6 +189,36 @@ message CreateVolumeRequest {
   // logged, etc.) by the CO.
   // This field is OPTIONAL.
   map<string, string> controller_create_secrets = 5;
+
+  // Contains all attributes of the source snapshot to create volume
+  // from. This field is OPTIONAL.
+  SourceSnapshot source_snapshot = 6;
+}
+
+// The information about a source snapshot.
+message SourceSnapshot {
+  // SnapshotSize contains `used_capacity_byes` which is the physical
+  // space consumed by the snapshot in bytes. `used_capacity_byes` may
+  // be variable over time. This field is OPTIONAL.
+  // If SnapshotSize is not set, it indicates this size is unknown.
+  SnapshotSize size = 1;
+
+  // Contains identity information for the created snapshot. This field
+  // is REQUIRED. The identity information will be used by the CO in
+  // subsequent calls to refer to the provisioned snapshot.
+  string id = 2;
+
+  // Identity information for the source volume. Note that create
+  // snapshot from snapshot is not supported here so the source has to
+  // be a volume. This field is REQUIRED.
+  string volume_id = 3;
```

Comment:
Is this more for validation that the volume is being created from a snapshot of a volume? Else why is it needed when creating a volume from a snapshot.

Creating a snap of a snap is not supported which is fine.